recipes/nuraft/all/patches/0001-cmake_patches.patch

```diff
@@ -0,0 +1,262 @@
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -26,41 +44,41 @@ endif()
+ 
+ 
+ # === Find ASIO ===
+-if (BOOST_INCLUDE_PATH AND BOOST_LIBRARY_PATH)
+-    # If Boost path (both include and library) is given,
+-    # use Boost's ASIO.
+-    message(STATUS "Boost include path: " ${BOOST_INCLUDE_PATH})
+-    message(STATUS "Boost library path: " ${BOOST_LIBRARY_PATH})
+-
+-    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -DUSE_BOOST_ASIO")
+-
+-    set(ASIO_INCLUDE_DIR ${BOOST_INCLUDE_PATH})
+-    set(LIBBOOST_SYSTEM "${BOOST_LIBRARY_PATH}/libboost_system.a")
+-
+-else ()
+-    # If not, ASIO standalone mode.
+-    FIND_PATH(ASIO_INCLUDE_DIR
+-              NAME asio.hpp
+-              HINTS ${PROJECT_SOURCE_DIR}/asio/asio/include
+-                    $ENV{HOME}/local/include
+-                    /opt/local/include
+-                    /usr/local/include
+-                    /usr/include)
+-
+-    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -DASIO_STANDALONE")
+-
+-endif ()
+-
+-if (NOT ASIO_INCLUDE_DIR)
+-    message(FATAL_ERROR "Can't find ASIO header files")
+-else ()
+-    message(STATUS "ASIO include path: " ${ASIO_INCLUDE_DIR})
+-endif ()
++#if (BOOST_INCLUDE_PATH AND BOOST_LIBRARY_PATH)
++#    # If Boost path (both include and library) is given,
++#    # use Boost's ASIO.
++#    message(STATUS "Boost include path: " ${BOOST_INCLUDE_PATH})
++#    message(STATUS "Boost library path: " ${BOOST_LIBRARY_PATH})
++#
++#    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -DUSE_BOOST_ASIO")
++#
++#    set(ASIO_INCLUDE_DIR ${BOOST_INCLUDE_PATH})
++#    set(LIBBOOST_SYSTEM "${BOOST_LIBRARY_PATH}/libboost_system.a")
++#
++#else ()
++#    # If not, ASIO standalone mode.
++#    FIND_PATH(ASIO_INCLUDE_DIR
++#              NAME asio.hpp
++#              HINTS ${PROJECT_SOURCE_DIR}/asio/asio/include
++#                    $ENV{HOME}/local/include
++#                    /opt/local/include
++#                    /usr/local/include
++#                    /usr/include)
++#
++#    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -DASIO_STANDALONE")
++#
++#endif ()
++
++#if (NOT ASIO_INCLUDE_DIR)
++#    message(FATAL_ERROR "Can't find ASIO header files")
++#else ()
++#    message(STATUS "ASIO include path: " ${ASIO_INCLUDE_DIR})
++#endif ()
+ 
+ 
+ # === Includes ===
+ include_directories(BEFORE ./)
+-include_directories(BEFORE ${ASIO_INCLUDE_DIR})
++#include_directories(BEFORE ${ASIO_INCLUDE_DIR})
+ include_directories(BEFORE ${PROJECT_SOURCE_DIR}/include)
+ include_directories(BEFORE ${PROJECT_SOURCE_DIR}/include/libnuraft)
+ include_directories(BEFORE ${PROJECT_SOURCE_DIR}/examples)
+@@ -213,6 +231,7 @@ if (NOT (DISABLE_SSL GREATER 0))
+     )
+ endif ()
+ 
++include_directories(BEFORE "${PROJECT_SOURCE_DIR}/include/libnuraft")
+ # === Source files ===
+ set(RAFT_CORE
+     ${ROOT_SRC}/asio_service.cxx
+@@ -240,6 +259,8 @@ set(RAFT_CORE
+     ${ROOT_SRC}/snapshot_sync_req.cxx
+     ${ROOT_SRC}/srv_config.cxx
+     ${ROOT_SRC}/stat_mgr.cxx
++    examples/in_memory_log_store.cxx
```

Comment:
Why is an example being included in the code library? 🤔

Reply:
Ah, it makes for a trivial test_package when recipe was written for private usage. Yes will remove from this recipe though and provide alternative solution.